src/script/descriptor.cpp

```diff
@@ -1163,17 +1163,36 @@ class WSHDescriptor final : public DescriptorImpl
     }
 };
 
+/** Represents the type of a node in taproot descriptor script tree */
+enum TRNodeType {
+    LEAF_SCRIPT,
+    NODE_HASH
+};
+
+/** A struct hold information on a node taproot descriptor script tree */
+struct TRNodeInfo {
```

Comment:
In  "descriptor: Add TRNodeInfo" 475981527f052f7c9709d1f3a90270770f4dc160

```suggestion
struct TaprootNode {
```